src/PhpSpec/CodeAnalysis/AccessInspectorInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Phpspec\CodeAnalysis;
+
+interface AccessInspectorInterface
```

Comment:
Yeah, that's where I was going, but it does mean that in order to support BC that hack in the Caller constructor would have to do the same decoration stuff. I think it's something to aim for in refactoring when you can make a BC break.

The existing solution will work well enough for MageSpec, it can just replace the MagicAwareAccessInspector for a MagicUnawareAccessInspector. Probably not named quite like that though.